bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/entity/generic/BusinessPartner.kt

```diff
@@ -92,4 +92,33 @@ class BusinessPartner(
     @Enumerated(EnumType.STRING)
     var parentType: BusinessPartnerType? = null
 
-) : BaseEntity()
+) : BaseEntity() {
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
For the `BusinessPartner` and `PostalAddress` I agree with Nico's opinion about overriding equals. Might make problems for JPA. Additionally it's dangerous to override equals but not hashCode.
Maybe better use a custom `hasRelevantChanges` function or something in the service.